cli/jdl.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2013-2020 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see https://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const chalk = require('chalk');
+const fs = require('fs');
+const https = require('https');
+const path = require('path');
+const cliUtils = require('./utils');
+const importJdl = require('./import-jdl');
+
+const { logger } = cliUtils;
+
+/**
+ * Add jdl extension to the file
+ */
+const toJdlFile = file => {
+    if (!path.extname(file)) {
+        return `${file}.jdl`;
+    }
+    return file;
+};
+
+const downloadFile = (url, filename) => {
+    return new Promise((resolve, reject) => {
+        logger.info(`Downloading file: ${url}`);
+        https
+            .get(url, response => {
+                if (response.statusCode !== 200) {
+                    return reject(new Error(`Error downloading ${url}: ${response.statusCode} - ${response.statusMessage}`));
+                }
+
+                logger.debug(`Creating file: ${path.join(filename)}`);
+                const fileStream = fs.createWriteStream(`${filename}`);
+                fileStream.on('finish', () => {
+                    fileStream.close(() => {
+                        resolve(filename);
```

Comment:
This doesn't see right. Filestream.close doesn't take any args. My VSCode also complains
![image](https://user-images.githubusercontent.com/1107223/82749593-7750b300-9daa-11ea-8792-606f12b2d797.png)

Reply:
It exists but I could not find a doc for WriteStream.close().
https://github.com/nodejs/node/blob/49f549e064e92c1db0392994d8a4636726d38a75/lib/internal/fs/streams.js#L461-L468

Reply:
But 'close' event is cleaner 👍